packages/color-scale-block/src/components/SquareWithColor.tsx

```diff
@@ -0,0 +1,116 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useDrag } from 'react-dnd';
+import { Button, ButtonSize, ButtonStyle, IconSize, IconTrashBin, Tooltip, merge, useCopy } from '@frontify/fondue';
+import { ItemDragState, TooltipAlignment, TooltipPosition } from '@frontify/fondue';
+import { SquareWithColorProps } from '../types';
+import { DragHandle } from './DragHandle';
+import { TooltipContent } from './TooltipContent';
+import { joinClassNames, toHex8String } from '@frontify/guideline-blocks-shared';
+
+export const SquareWithColor = ({
+    id,
+    index,
+    width,
+    height,
+    isFirst,
+    isLast,
+    currentColor,
+    onResizeStart,
+    calculateLeftPosition,
+    isEditing,
+    deleteColor,
+    listId,
+    setCurrentlyDraggedColorId,
+    isDragging,
+}: SquareWithColorProps) => {
+    const handleDrag = () => {
+        if (isDragging && currentColor !== null) {
+            setCurrentlyDraggedColorId(currentColor.id ?? null);
+        }
+    };
+
+    const handleDragEnd = () => {
+        setCurrentlyDraggedColorId(null);
+    };
+
+    const [{}, drag] = useDrag({
+        item: currentColor,
+        collect: (monitor) => {
+            return {
+                componentDragState: monitor.isDragging() ? ItemDragState.Dragging : ItemDragState.Idle,
+            };
+        },
+        type: listId,
+        canDrag: isEditing,
+    });
+
+    const { copy, status } = useCopy();
+
+    const rgbaValues = [
+        currentColor.red,
+        currentColor.green,
+        currentColor.blue,
+        ...(currentColor.alpha ? [currentColor.alpha / 255] : []),
+    ];
+
+    const backgroundColor = `rgba(${rgbaValues.join(',')})`;
+    const hexColor = currentColor ? toHex8String(currentColor) : '';
+
+    const firstElementClasses = 'tw-pl-[1px] tw-pr-[1px] tw-rounded-tl tw-rounded-bl';
+    const lastElementClasses = 'tw-rounded-tr tw-rounded-br-[3px]';
+
+    return (
+        <div
+            style={{
+                width: `${width}px`,
+                left: `${calculateLeftPosition(index, width)}px`,
+                height,
+            }}
+            className="hover:tw-z-30 tw-overflow-x-visible tw-overflow-y-hidden tw-pb-8 tw-inline-block"
+        >
+            {isEditing ? <DragHandle index={index} currentColor={currentColor} onResizeStart={onResizeStart} /> : <></>}
+
+            <div
+                ref={drag}
+                draggable
+                onDrag={handleDrag}
+                onDragEnd={handleDragEnd}
+                style={{
+                    height,
+                    backgroundColor,
+                }}
+                className={`${merge([
+                    isFirst ? firstElementClasses : '',
+                    isLast ? lastElementClasses : '',
+                    '[&>div]:tw-h-full tw-group tw-relative tw-w-full tw-h-full tw-overflow-y-hidden tw-overflow-x-visible tw-border-none hover:tw-border hover:tw-border-black',
```

Comment:
```suggestion
                    '[&>div]:tw-h-full tw-group tw-relative tw-w-full tw-h-full tw-overflow-y-hidden tw-overflow-x-visible tw-border-none hover:tw-border hover:tw-border-black',
                    isFirst ? firstElementClasses : '',
                    isLast ? lastElementClasses : '',
```
Static first and conditional second is always a good (you might override some classes by mistake otherwise)